doc-content/enterprise-only/openshift/hacep-con.adoc

```diff
@@ -16,3 +16,7 @@ In this reference implementation, you can scale the pod with the processing code
 The election of the leader is implemented with Kubernetes ConfigMaps. Coordination of the leader with other replicas is performed with exchanged messages through Kafka. The leader is always the first to process an event. When processing is complete, the leader notifies other replicas. A replica that is not the leader starts executing an event only after it has been completely processed on the leader. 
 
 When a new replica joins the cluster, this replica requests a snapshot of the current {PRODUCT_DROOLS} session from the leader. The leader can use a recent existing snapshot if one is available in a Kafka topic. If a recent snapshot is not available, the leader produces a new snapshot on demand. After receiving the snapshot, the new replica deserializes it and eventually executes the last events not included in the snapshot before starting to process new events in coordination with the leader.
+
+With the default implementation method, the service is built into the HA CEP server as a fat KJAR. In this case, you must build and deploy the server again to change the version of the service. The content of the working memory is lost when you switch to the new version. For instructions about the default implementation method, see <<hacep-server-proc>>.
+
+If you want to upgrade versions of the service without losing the content of the working memory, you can use an alternate implementation method and provide the KJAR and all dependencies in a Maven repository. You can use an `UpdateKJarGAV` call from the client code to trigger deployment of a new KJAR version. This call is processed by the leader and then other replicas, and each of the pods then loads the new KJAR. The contents of the working memory remain in place. For instructions about this implementation method, see <<hacep-server-maven-proc>>.
```

Comment:
Minimalism again: Remove "If you want" and "you can"

To upgrade versions of the service without losing the content of the working memory, use an alternate implementation method and provide the KJAR and all dependencies in a Maven repository. Use an `UpdateKJarGAV` call from the client code to trigger deployment of a new KJAR version. This call is processed by the leader and then other replicas, and each of the pods then loads the new KJAR. The contents of the working memory remain in place. For instructions about this implementation method, see <<hacep-server-maven-proc>>.